config.go

```diff
@@ -247,3 +262,49 @@ func isValidIAM(arn *string, client *alks.Client) bool {
 func splitBy(r rune) bool {
 	return r == ':' || r == '/'
 }
+
+func generateNewClient(c *Config, cident *sts.GetCallerIdentityOutput, client *alks.Client) (*alks.Client, error) {
+
+	// 3. Create account string
+	newAccDetail := c.Account + "/ALKS" + c.Role
+
+	// Calling for the same account; fine.
+	if strings.Contains(newAccDetail, client.AccountDetails.Account) {
+
+		// check if the user is using a assume-role IAM admin session or MI.
+		if isValidIAM(cident.Arn, client) != true {
+			return nil, errors.New("Looks like you are not using ALKS IAM credentials. This will result in errors when creating roles. \n " +
+				"Note: If using ALKS CLI to get credentials, be sure to use the '-i' flag. \n Please see https://coxautoinc.sharepoint.com/sites/service-internal-tools-team/SitePages/ALKS-Terraform-Provider---Troubleshooting.aspx for more information.")
+		}
+
+		client.SetUserAgent(fmt.Sprintf("alks-terraform-provider-%s", getPluginVersion()))
+
+		log.Println("[INFO] ALKS Client configured")
+
+		return client, nil
+	} else {
+
+		// 4. Alright, new credentials needed - swap em out.
+		client.AccountDetails.Account = newAccDetail
+		client.AccountDetails.Role = c.Role
+
+		newCreds, _ := client.CreateIamSession()
```

Comment:
This line can fail, we should check for an error here and return early if we cannot generate keys.
```suggestion
		newCreds, err := client.CreateIamSession()
		if err != nil {
			return nil, err
		}
```